backends/tfhe-cuda-backend/build.rs

```diff
@@ -51,6 +54,53 @@ fn main() {
         println!("cargo:rustc-link-lib=cudart");
         println!("cargo:rustc-link-search=native=/usr/lib/x86_64-linux-gnu/");
         println!("cargo:rustc-link-lib=stdc++");
+
+        let header_path = "wrapper.h";
+        println!("cargo:rerun-if-changed={}", header_path);
+
+        let out_path = PathBuf::from("src").join("bindings.rs");
+
+        // Check modification times
+        let header_modified = fs::metadata(header_path).unwrap().modified().unwrap();
+        let bindings_modified = if out_path.exists() {
+            fs::metadata(&out_path).unwrap().modified().unwrap()
+        } else {
+            std::time::SystemTime::UNIX_EPOCH // If bindings file doesn't exist, consider it older
+        };
+
+        // Regenerate bindings only if header has been modified
+        if header_modified > bindings_modified {
+            let bindings = bindgen::Builder::default()
+                .header(header_path)
+                .clang_arg("-x")
+                .clang_arg("c++")
+                .clang_arg("-std=c++17")
+                .clang_arg("-I/usr/include")
+                .clang_arg("-I/usr/local/include")
+                .ctypes_prefix("ffi")
+                .raw_line("use crate::ffi;")
+                .generate()
+                .expect("Unable to generate bindings");
+
+            bindings
+                .write_to_file(&out_path)
+                .expect("Couldn't write bindings!");
+
+            // Read, modify, and write back the generated file
+            let mut bindings_content =
+                fs::read_to_string(&out_path).expect("Couldn't read the generated bindings file");
+
+            // Prepend the `#![allow(...)]` attributes
+            let custom_attributes = "\
+#![allow(non_upper_case_globals)]\n\
+#![allow(non_snake_case)]\n\
+#![allow(improper_ctypes)]\n\
+#![allow(warnings)]\n";
+
+            bindings_content = format!("{}{}", custom_attributes, bindings_content);
+            fs::write(&out_path, bindings_content)
+                .expect("Couldn't write modified bindings to file!");
+        }
```

Comment:
another way to go about this is to have the attributes as outer attributes on the mod.

```rust
#[allow(non_upper_case_globals)]
#[allow(non_snake_case)]
#[allow(improper_ctypes)]
#[allow(warnings)]
pub mod bindings;
```

this way no need to keep manual fixes for the lints in the build.rs, we can update them outside in the code base more easily, the above should work hopefully but attributes don't always play nice so maybe it won't

the improper_ctypes is not really reassuring though ? and allowing warnings too, I know it's auto generated but still, what are the warnings thrown by clippy ?

Reply:
I'll try again

Reply:
So I can do as you say and it works, but I have to keep `#[allow(warnings)]` otherwise I get all the case issues printed as warnings. Things like:
```warning: type `uint_least32_t` should have an upper camel case name                                                                                                                                        
   --> backends/tfhe-cuda-backend/src/bindings.rs:230:10                                                                                                                                                   
    |                                                                                                                                                                                                      
230 | pub type uint_least32_t = __uint_least32_t;                                                                                                                                                          
    |          ^^^^^^^^^^^^^^ help: convert the identifier to upper camel case: `UintLeast32T`    
    ```

Reply:
isn't there a specific allow for this one ?

Reply:
Yes there is, it is #[allow(clippy::upper_case_acronyms)], but actually only #[allow(warnings)] seems to have an effect at all :thinking:

Reply:
😵‍💫